minio/minio
# Label the notification target metrics by their target IDs

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

This patch adds the targetID to the existing notification target metrics and deprecates the current target metrics which points to the overall event notification subsystem

## Motivation and Context
To make the notification target metrics classified by their target ids

## How to test this PR?
- Set up bucket notifications
- Check prometheus metrics, you can see new target level metrics

For eg,

```
➜  minio git:(notification-target-metrics) mc admin prometheus metrics myminio | grep notify
# HELP minio_notify_current_send_in_progress Number of concurrent async Send calls active to all targets (deprecated, please use 'target_current_send_in_progress' instead)
# TYPE minio_notify_current_send_in_progress gauge
minio_notify_current_send_in_progress{server="127.0.0.1:9000"} 0
# HELP minio_notify_events_errors_total Events that were failed to be sent to the targets (deprecated, please use 'target_failed_events' instead)
# TYPE minio_notify_events_errors_total counter
minio_notify_events_errors_total{server="127.0.0.1:9000"} 0
# HELP minio_notify_events_sent_total Total number of events sent to the targets (deprecated, please use 'target_total_events' instead)
# TYPE minio_notify_events_sent_total counter
minio_notify_events_sent_total{server="127.0.0.1:9000"} 50
# HELP minio_notify_events_skipped_total Events that were skipped to be sent to the targets due to the in-memory queue being full
# TYPE minio_notify_events_skipped_total counter
minio_notify_events_skipped_total{server="127.0.0.1:9000"} 0
# HELP minio_notify_target_current_send_in_progress Number of concurrent async Send calls active to the target
# TYPE minio_notify_target_current_send_in_progress gauge
minio_notify_target_current_send_in_progress{server="127.0.0.1:9000",target_id="1",target_name="webhook"} 0
# HELP minio_notify_target_failed_events Number of events failed to be sent (or) queued to the target
# TYPE minio_notify_target_failed_events counter
minio_notify_target_failed_events{server="127.0.0.1:9000",target_id="1",target_name="webhook"} 0
# HELP minio_notify_target_queue_length Number of events currently staged in the queue_dir configured for the target
# TYPE minio_notify_target_queue_length gauge
minio_notify_target_queue_length{server="127.0.0.1:9000",target_id="1",target_name="webhook"} 0
# HELP minio_notify_target_total_events Total number of events sent (or) queued to the target
# TYPE minio_notify_target_total_events counter
minio_notify_target_total_events{server="127.0.0.1:9000",target_id="1",target_name="webhook"} 50

```


## Types of changes

## Checklist:
